### PR TITLE
Post processor scorer

### PR DESCRIPTION
**Pull request by Proca-Consul**
Post processor refactored (and introduced new class in json-parser) [Ivan ft. Dom]

I have also written tests for both the components of the post processor.

WARNING: contains profusion of higher order Java 8 functions

### CODE REVIEW

File: lib/json_parser/src/main/scala/json_parser/AggregatorOutput.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import play.api.libs.json._
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+
+/**
+  * Created by Consul on 25/10/2016.
+  */
```
Comment (AbraaoMota) [inline lib/json_parser/src/main/scala/json_parser/AggregatorOutput.scala]: for consistency, remove these

File: backend/postprocessor/src/main/java/postprocessor/PostProcessor.java [hunk 2]
```diff
@@ -1,31 +1,47 @@
...
         String language = args[0];
-        List<String> inputJsonPaths = new ArrayList<>();
+        List<String> inputPaths = new ArrayList<>();
         int i;
         for (i = 1; i < args.length - 1; i++) {
-            inputJsonPaths.add(args[i]);
+            inputPaths.add(args[i]);
         }
-        String outputJsonPath = args[i + 1];
+        String outputJsonPath = args[i];
```
Comment (giuliojiang) [inline backend/postprocessor/src/main/java/postprocessor/PostProcessor.java]: args[args.length - 1] ?

File: backend/postprocessor/pom.xml [hunk 3]
```diff
@@ -19,6 +19,11 @@
       <version>1.1.1</version>
```
Comment (helicopter88) [inline backend/postprocessor/pom.xml]: You should remove json_simple from this file

File: backend/postprocessor/src/main/java/postprocessor/PostProcessorUtils.java [hunk 4]
```diff
@@ -1,31 +1,30 @@
 package postprocessor;
 
+import json_parser.MicroServiceOutput;
+import json_parser.MicroServiceOutputParser;
 import org.json.simple.JSONObject;
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
```
Comment (helicopter88) [inline backend/postprocessor/src/main/java/postprocessor/PostProcessorUtils.java]: these imports should be removed

**Comment by helicopter88** [comment]
Why was this merged with some comments still left?

**Comment by AbraaoMota** [comment]
Whoops I think you gave the OK on friday afternoon and I was in a rush so I overlooked them sorry. I can make a PR to fix those

**Pull request merged**